services/ble/Gap.hpp

```diff
@@ -328,7 +330,7 @@ namespace services
         void StateChanged(GapState state) override;
 
         // Implementation of GapCentral
-        void Connect(hal::MacAddress macAddress, GapDeviceAddressType addressType) override;
+        void Connect(hal::MacAddress macAddress, GapDeviceAddressType addressType, infra::Duration initiatingTimeout) override;
```

Comment:
Will disconnect cancel a initiating state, or does the user have to wait for the timeout?

Reply:
No, a disconnect function will fail with 0x3c (command disallowed), but no event will be triggered towards the application. To my opinion a disconnect should **not** cancel a connection, because it should act after the connection is established.

Reply:
I will add a new function to Cancel a connection request.